carma_wm/src/WMListenerWorker.cpp

```diff
@@ -76,7 +78,21 @@ void WMListenerWorker::mapUpdateCallback(const autoware_lanelet2_msgs::MapBinCon
     }
     else
     {
-      world_model_->getMutableMap()->update(parent_llt, pair.second);
+      auto factory_pcl = lanelet::RegulatoryElementFactory::create(pair.second.get()->attribute(lanelet::AttributeName::Subtype).value(),
+                                                            std::const_pointer_cast<lanelet::RegulatoryElementData>(pair.second.get()->constData()));
+      
+      // we should extract general regem to specific type of regem the geofence specifies
+      if (pair.second.get()->attribute(lanelet::AttributeName::Subtype).value() == lanelet::PassingControlLine::RuleName)
```

Comment:
Lets make these checks into a switch to make it easier to read and update in the future.